calico-vpp-agent/config/config.go

```diff
@@ -139,12 +139,19 @@ func getEnvValue(str string) string {
 }
 
 func fetchHostMtu() (mtu int, err error) {
+	vppManagerInfo := &vppmanagerconfig.VppManagerInfo{}
 	for i := 0; i < 20; i++ {
-		dat, err := ioutil.ReadFile(VppManagerLinuxMtu)
+		dat, err := ioutil.ReadFile(VppManagerInfoFile)
 		if err == nil {
-			idx, err := strconv.ParseInt(strings.TrimSpace(string(dat[:])), 10, 32)
-			if err == nil && idx != -1 {
-				return int(idx), nil
+			err2 := json.Unmarshal(dat, vppManagerInfo)
```

Comment:
This actually is called before `waitForVppManager` so before setting the `common.VppManagerInfo` variable

Reply:
But if I understand correctly, this is only to set `config.HostMtu` which in turn is only used in `computePodMtu`
We should probably remove `config.HostMtu` and directly use the value in `vppManagerInfo` (also probably take the min of all MTUs to define the podMtu in case of multiple uplinks)